.circleci/config.yml

```diff
@@ -0,0 +1,104 @@
+    # apparently.  We cannot set the working directory or environment variables by using
+    # other environment variables (although the working directory can use the `~` character,
+    # but environment variables cannot), so to avoid having to override the GOPATH for every
+    # run command, just hard code in the directory to be CircleCI expects it to be.
+    working_directory: /home/circleci/.go_workspace/src/github.com/docker/swarmkit
+    environment:
+      # Needed to install go
+      OS: linux
+      ARCH: amd64
+      GOVERSION: 1.8.7
+      # Needed to install protoc
+      PROTOC: https://github.com/google/protobuf/releases/download/v3.5.0/protoc-3.5.0-linux-x86_64.zip
+
+    # Note(cyli): We create a tmpfs mount to be used for temporary files created by tests
+    # to mitigate the excessive I/O latencies that sometimes cause the tests to fail.
+    # See https://github.com/docker/swarmkit/pull/2254.
+
+    # There is no way to mount tmpfs volumes in the docker executor, so we are using
+    # the machine executor. However, this incur a performance overhead
+    # (https://discuss.circleci.com/t/using-docker-compose-in-2-0/9492/4)
+    # and in the future could incur additional pricing changes
+    # (https://circleci.com/docs/2.0/executor-types/#using-machine).
+
+    # One possible hack is the following:
+
+    # /dev/shm in the container is tmpfs although files in /dev/shm are not executable.
+    # If we specify TMPDIR=/dev/shm, /dev/shm will be used by our tests, which call
+    # ioutil.TempDir/ioutil.TempFile, to write temporary files.
+    # We can also specify GOTMPDIR=/tmp or some other non-tmpfs directory
+    # (see https://golang.org/doc/go1.10#goroot) - this is the directory in which the
+    # go tool itself will put temporarily compiled test executables, etc.
+
+    # However, using this hack still resulted in occasional WAL test failures,
+    # so it seems like it does not work, or there may be some other failure.
+    # It may be something to explore again if the penalty for using the machine
+    # executor becomes unacceptable.
+
+    machine: true
+
+    steps:
+    - checkout
+
+    # This would not be needed if we used a golang docker image
+    - run:
+        name: Install go
+        command: |
+            sudo rm -rf /usr/local/go
+            curl -fsSL -o "$HOME/go.tar.gz" "https://storage.googleapis.com/golang/go$GOVERSION.$OS-$ARCH.tar.gz"
+            sudo tar -C /usr/local -xzf "$HOME/go.tar.gz"
+
+    - run:
+        name: Output debugging information
+        command: |
+            go version
+            env
+
+    - run:
+        name: Install protoc
+        command: |
+            curl -fsSL -o "$HOME/$(basename $PROTOC)" "$PROTOC"
+            unzip -o "$HOME/$(basename $PROTOC)" -d "$HOME"
+            sudo cp -R "$HOME/include/google" /usr/local/include
+            sudo chmod 777 -R /usr/local/include/google
```

Comment:
Note that the two lines above are actually part of https://github.com/docker/swarmkit/pull/2349/files#diff-29944324a3cbf9f4bd0162dfe3975d88R36, and leaked through due to the rename/move of this file.

Don't think they should really bother if we decide to not include that PR in 17.06